src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -43,15 +44,25 @@ public void addMessage(String message) {
         if (messages == null) {
             messages = new ArrayList<String>();
         }
-        messages.add(message);
+        if (StringUtils.isNotBlank(message) && isNotDuplicateMessage(message)) {
+            messages.add(message);
+        }
+    }
+    
+    private boolean isNotDuplicateMessage(String message) {
+        for (String currentMessage : getMessages()) {
```

Comment:
The "messages" field is initialized to an empty list in the default constructor, or to an explicit list in the other constructor. Unless other code is trying to pass null lists into the constructors or setters, then I think this code is fine as-is.